modules/scala-support/src/main/scala/dev/guardrail/generators/scala/circe/CirceRefinedProtocolGenerator.scala

```diff
@@ -23,9 +23,15 @@ object CirceRefinedProtocolGenerator {
   def apply(circeRefinedVersion: CirceRefinedModelGenerator): ProtocolTerms[ScalaLanguage, Target] =
     fromGenerator(CirceProtocolGenerator.withValidations(circeRefinedVersion.toCirce, applyValidations _))
 
-  def applyValidations(tpe: Type, prop: Tracker[Schema[_]]): Target[Type] = {
+  def applyValidations(tpe: Type, prop: Tracker[Schema[_]], className: String): Target[Type] = {
     import scala.meta._
     tpe match {
+      case t"String" =>
+        val pattern = prop.downField("pattern", _.getPattern).unwrapTracker
+        Target.pure(pattern.fold(tpe){ pat =>
+          val refined = s"""_root_.eu.timepit.refined.string.MatchesRegex[$className.`"$pat"`.T]""".parse[Type].get
```

Comment:
I'm surprised this unquotes

Reply:
Also, when you get to the point of `Target.raiseException`, please consider `prop.downField("pattern", _.getPattern).history` in your error message